### ISSUE TITLE
Setting the supported locales dynamically

### ISSUE THREAD

**Issue by jbroadway** [camsjams/i18n-unity-csharp#2]
First, I'm very happy I found this today! I implemented i18n support in my project in under an hour :)

I'm wondering if there's a reason the supported locales is a [private property](https://github.com/MoonGateLabs/i18n-unity-csharp/blob/30962a0b15fe9a07d137f96afdaf534c4af578ca/src/I18nUnity/Mgl/I18n.cs#L14)? It looks like we have to modify the source file every time we add a new supported language, but it would be nice to be able to leave the original code as is and modify the list dynamically.

**Comment by GabLeRoux** [comment]
I agree with you, this should be at least `protected` so we can extend and add new languages on the go. I really like this library too and I found it very handy combined with [weblate](https://weblate.org) as the translation backend.

I suppose you could send a PR but I don't know if this is still maintained. It did not change in two years but I offer my help here and will definitely look into this and review changes.

I'm using it in unity but I've copied the files directly in my projects instead of actually using it as a dependency to fit my needs. I didn't want my team to mess with submodules ;)

**Comment by camsjams** [comment]
I still support this library, and would be happy to discuss this improvement.

As @GabLeRoux says (and same as how we use it) we just copy this file directly into projects as it just works a little easier that way.

That being said, changing this to a `protected` property seems perfectly logical.

**Comment by jbroadway** [comment]
Thanks for the quick replies! For myself, I prefer subclassing over modifying directly in case an update comes out and another dev on our team copies that updates into our project without realizing they've overwritten our previous modifications.

I've submitted a PR that makes the changes needed and tested that subclassing works to override the locales for me. Here's what I needed to do:

```C#
namespace MyApp {
	public class I18n : Mgl.I18n {
		protected static readonly I18n instance = new I18n();
		
		protected static string[] locales = new string[] { "en-US", "fr-FR", "es-ES", "de-DE" };
		
		public static I18n Instance {
			get {
				return instance;
			}
		}
	}
}
```

Now I just have to replace `using Mgl` with `using MyApp` and I've added a new locale.

**Comment by GabLeRoux** [comment]
Awesome, I'll update the readme tonight with the instructions you provided. Definitely makes it easier to use and update. :shipit:

**Comment by camsjams** [comment]
Looks good! Thanks you two!

**Comment by GabLeRoux** [comment]
@jbroadway I know it's been a while, but I worked with this today, and I realized that overriding `locales` here doesn't work as expected. `locales` is *`static`* so the above example doesn't exactly override the `locales` inside base class. In other words, static belongs to the class and not the object so above code is only hiding it, oups!

When I tested this, it worked by _coincidence_ as I used `fr_FR` + `en_US`, but I tried with `fr_CA` instead and now it's failing with `Missing: locale [fr_CA] not found in supported list` where `fr_CA` comes from my own class, but `locales` contains base class values.

Quick workaround is still to edit base class directly which is easier, but it doesn't feel right, I'll see if I can do something about it, I should have tested this a bit more, sorry.

**Comment by lux** [comment]
Ah you're right! Hmm... What about a `SetLocales(locales)` method, or adding a `locales` option to `I18n.Configure()`?

**Comment by GabLeRoux** [comment]
I think it's the way to go, but this implies changing a few parts of the code that are currently `static`. If we add a `SetLocales(locales)`, we'll also need a `GetLocales()`.

We'll need to convert
```csharp
protected static string[] locales = new string[] { "en-US", "fr-FR", "es-ES" };
```
back to private non static:
```csharp
string[] locales = new string[] { "en-US", "fr-FR", "es-ES" };
```

And change the way [it's used inside](https://github.com/MoonGateLabs/i18n-unity-csharp/blob/39ab086dc85b2cf0d501b0fa4943291547147f09/src/I18nUnity/Mgl/I18n.cs#L39). Example, we'd have to remove `static` from:
```csharp
 static void InitConfig()
        {
            if (locales.Contains(_currentLocale))
{
```

I'm not convinced that using `static` almost everywhere makes the library easier to use. This is cascading quickly, `InitConfig` is used in a few other `static` methods too.

**Comment by camsjams** [comment]
@jbroadway @lux Thanks to work from @shinji-yoshida we have a pretty good solution for part of the issue
